File: rileylink/src/main/java/info/nightscout/androidaps/plugins/pump/common/hw/medlink/ble/command/BleBolusCommand.java
package info.nightscout.androidaps.plugins.pump.common.hw.medlink.ble.command;

import info.nightscout.androidaps.logging.AAPSLogger;
import info.nightscout.androidaps.logging.LTag;
import info.nightscout.androidaps.plugins.pump.common.hw.medlink.ble.MedLinkBLE;
import info.nightscout.androidaps.plugins.pump.common.hw.medlink.service.MedLinkServiceData;

/**
 * Created by Dirceu on 24/03/21.
 */
public  class BleBolusCommand extends BleCommand {




    public BleBolusCommand(AAPSLogger aapsLogger,
                           MedLinkServiceData medlinkServiceData, Runnable runnable) {
        super(aapsLogger, medlinkServiceData, runnable);
    }

    public BleBolusCommand( AAPSLogger aapsLogger, MedLinkServiceData medlinkServiceData) {
        super(aapsLogger, medlinkServiceData);
    }

    @Override public void characteristicChanged(String answer, MedLinkBLE bleComm,
                                                String lastCommand) {
        super.characteristicChanged(answer, bleComm,lastCommand);
        if (answer.trim().contains("set bolus")) {
            aapsLogger.info(LTag.PUMPBTCOMM,pumpResponse.toString());
            bleComm.completedCommand();
        }

    }
}

File: rileylink/src/main/java/info/nightscout/androidaps/plugins/pump/common/hw/medlink/ble/command/BleBolusCommand.kt
package info.nightscout.androidaps.plugins.pump.common.hw.medlink.ble.command

import info.nightscout.androidaps.plugins.pump.common.MedLinkPumpPluginAbstract
import info.nightscout.androidaps.plugins.pump.common.hw.medlink.ble.MedLinkBLE
import info.nightscout.androidaps.plugins.pump.common.hw.medlink.service.MedLinkServiceData
import info.nightscout.shared.logging.AAPSLogger
import info.nightscout.shared.logging.LTag

/**
 * Created by Dirceu on 24/03/21.
 */
class BleBolusCommand : BleActivePumpCommand {
    constructor(aapsLogger: AAPSLogger,
                medLinkServiceData: MedLinkServiceData,
                medLinkPumpPluginAbstract: MedLinkPumpPluginAbstract
    ) : super(aapsLogger, medLinkServiceData, medLinkPumpPluginAbstract)

    override fun characteristicChanged(answer: String, bleComm: MedLinkBLE,
                                       lastCommand: String) {
        aapsLogger.info(LTag.PUMPBTCOMM, answer)
        aapsLogger.info(LTag.PUMPBTCOMM, lastCommand)
        if (answer.trim { it <= ' ' }.contains("set bolus")) {
            aapsLogger.info(LTag.PUMPBTCOMM, pumpResponse.toString())
            bleComm.completedCommand()
        } else {
            super.characteristicChanged(answer, bleComm, lastCommand)
        }
    }
}